notifications_utils/template.py

```diff
@@ -66,40 +73,50 @@ def values(self, value):
             )
 
     @property
-    def renderer(self):
-        return self._renderer
-
-    @renderer.setter
-    def renderer(self, value):
-        if value:
-            self._renderer = value
-        elif self.template_type == 'sms':
-            self._renderer = SMSPreview(
-                prefix=self._prefix,
-                sender=self._sms_sender
-            )
-        elif self.template_type == 'letter':
-            self._renderer = LetterPreview()
-        elif self.template_type == 'email' or not self.template_type:
-            self._renderer = EmailPreview()
+    def placeholders(self):
+        return Field(self.content).placeholders
 
     @property
-    def rendered(self):
-        return self.renderer(self.to_dict())
+    def missing_data(self):
+        return list(
+            placeholder for placeholder in self.placeholders
+            if self.values.get(placeholder) is None
+        )
 
     @property
-    def placeholders(self):
-        return Field(self.subject or '').placeholders | Field(self.content).placeholders
+    def additional_data(self):
+        return self.values.keys() - self.placeholders
 
-    @property
-    def content_count(self):
-        return len(self.rendered.encode(self.encoding))
+    def get_raw(self, key, default=None):
+        return self._template.get(key, default)
+
+    def compare_to(self, new):
+        return TemplateChange(self, new)
+
+
+class SMSMessageTemplate(Template):
+
+    def __init__(
+        self,
+        template,
+        values=None,
+        prefix=None,
+        sender=None
+    ):
+        self.prefix = prefix
+        self.sender = sender
+        super().__init__(template, values)
+
+    def __str__(self):
+        return Take.as_field(
+            self.content, self.values
+        ).then(
+            add_prefix, self.prefix if not self.sender else None
+        ).as_string.strip()
 
     @property
-    def sms_fragment_count(self):
-        if self.template_type != 'sms':
-            raise TypeError("The template needs to have a template type of 'sms'")
-        return get_sms_fragment_count(self.content_count)
+    def content_count(self):
+        return len(str(self).encode(self.encoding))
 
     @property
     def content_too_long(self):
```

Comment:
Yeah, the API was doing this comparison itself. Removed in: 7997fc7d7c30a4f1f26280997404ab73837f50a7